File: DSA/N_PARTH_PANDEY_590012761/Day17Question3.py
def integer_sqrt(num):
    if num < 2:
        return num  
    
    left, right = 1, num
    while left <= right:
        mid = (left + right) // 2
        if mid * mid == num:
            return mid
        elif mid * mid < num:
            left = mid + 1
        else:
            right = mid - 1
    return right  


num = int(input("Enter any positive square number: "))
print(integer_sqrt(num))  

File: DSA/Rahul_Mankani_590014725/Day17_Question3.c
#include <stdio.h>

int main() 
{
    int num;
    int result = 0;
    printf("Enter a positive number: ");
    scanf("%d", &num);
    if (num < 0) 
    {
        printf("Invalid input! Enter a positive number.\n");
        return 1;
    }
    int left = 0;
    int right = num;
    while (left <= right) 
    {
        int mid = left + (right - left) / 2;
        long square = (long)mid * mid;
        if (square == num) 
        {
            result = mid;
            break;
        }
        else if (square < num) 
        {
            result = mid;
            left = mid + 1;
        }
        else 
        {
            right = mid - 1;
        }
    }
    printf("Integer square root of %d is %d.\n", num, result);
    return 0;
}
